### PR TITLE
Unit tests for model package

### PR DESCRIPTION
**Pull request by ryansteakley**
Description of changes:
Sdk.go coverage: 81.1%
overall coverage: 56.21 %
custom code coverage: > 85%

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/resource/model_package/testdata/sdkapi/describe/deleting.json [hunk 1]
```diff
@@ -0,0 +1,87 @@
+{
+    "ApprovalDescription": null,
```
Comment (mbaijal) [inline pkg/resource/model_package/testdata/sdkapi/describe/deleting.json:2]: is this describe really different from a created successfully describe ?

File: pkg/resource/model_package/testdata/v1alpha1/create/desired/success_before_create.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+apiVersion: sagemaker.services.k8s.aws/v1alpha1
```
Comment (mbaijal) [inline pkg/resource/model_package/testdata/v1alpha1/create/desired/success_before_create.yaml:1]: nit: not sure what the filename means - success before create for a desired spec ?

File: pkg/resource/model_package/testdata/test_suite.yaml [hunk 3]
```diff
@@ -1,17 +1,251 @@
...
+        given:
+          desired_state: "v1alpha1/create/desired/success_before_create.yaml"
+          svc_api:
+            - operation: CreateModelPackageWithContext
+              output_fixture: "sdkapi/create/create_success.json"
+        invoke: Create
+        expect:
+          latest_state: "v1alpha1/create/observed/success_after_create.yaml"
+          error: nil
+      - name: "Create=ValidVariation"
```
Comment (mbaijal) [inline pkg/resource/model_package/testdata/test_suite.yaml:29]: Better than Create=Valid2 but still can we add a better name to know why a variation needs to be tested - is it more fields ?

File: pkg/resource/model_package/testdata/test_suite.yaml [hunk 4]
```diff
@@ -1,17 +1,251 @@
...
+        description: "Testing readone right after create, the status should be in Pending with Resource synced being false."
+        given: 
+          desired_state: "v1alpha1/readone/desired/pending_after_create.yaml"
+          svc_api:
+            - operation: DescribeModelPackageWithContext
+              output_fixture: "sdkapi/describe/describe_pending_after_create.json"
+        invoke: ReadOne
+        expect:
+          latest_state: "v1alpha1/readone/observed/pending_after_create.yaml"
+      - name: "ReadOne=Deleting"
```
Comment (mbaijal) [inline pkg/resource/model_package/testdata/test_suite.yaml:96]: Nice, I don't suppose other resources have this one

File: pkg/resource/model_package/testdata/test_suite.yaml [hunk 5]
```diff
@@ -1,17 +1,251 @@
...
+          svc_api:
+            - operation: DescribeModelPackageWithContext
+              output_fixture: "sdkapi/describe/describe_success.json"
+        invoke: ReadOne
+        expect:
+          latest_state: "v1alpha1/readone/observed/conditions_clear_on_success.yaml"
+  - name: "Model package update tests"
+    description: "Testing the Update operation"
+    scenarios:
+      - name: "Update=NotSupported"
```
Comment (mbaijal) [inline pkg/resource/model_package/testdata/test_suite.yaml:149]: ++, good to have !

**Review by mbaijal — APPROVED** [review]

File: pkg/resource/model_package/testdata/v1alpha1/create/desired/success_before_create.yaml (hunk 2 above)
Comment (ryansteakley) [reply pkg/resource/model_package/testdata/v1alpha1/create/desired/success_before_create.yaml]: miss on my part renamed to before_create

File: pkg/resource/model_package/testdata/test_suite.yaml (hunk 3 above)
Comment (ryansteakley) [reply pkg/resource/model_package/testdata/test_suite.yaml]: renamed to clarify the inputs

File: pkg/resource/model_package/testdata/test_suite.yaml (hunk 4 above)
Comment (ryansteakley) [reply pkg/resource/model_package/testdata/test_suite.yaml:96]: Haven't seen other resources with pending state :)

**Review by mbaijal — APPROVED** [review]

**Pull request merged**